controllers/vrg_volrep.go

```diff
@@ -170,8 +171,8 @@ func (v *VRGInstance) reconcileVRAsSecondary(pvc *corev1.PersistentVolumeClaim,
 func (v *VRGInstance) isPVCReadyForSecondary(pvc *corev1.PersistentVolumeClaim, log logr.Logger) bool {
 	const ready bool = true
 
-	// If PVC is not being deleted, it is not ready for Secondary
-	if pvc.GetDeletionTimestamp().IsZero() {
+	// If PVC is not being deleted, it is not ready for Secondary, unless action is failover
+	if v.instance.Spec.Action != ramendrv1alpha1.VRGActionFailover && pvc.GetDeletionTimestamp().IsZero() {
```

Comment:
In what case, we move the VRG to secondary but the workload is not deleted? IOW, what are we solving?
I think we still can't delete the VRG even if we move it to secondary because `skipPVC` will skip because the PVC is not deleted.

Reply:
Looked at `skipPVC`, in the delete workflow if PVC is not yet protected by VRG [`reconcileVRsForDeletion`](https://github.com/RamenDR/ramen/blob/acf2e04a81e80a511d387a09e55a7245134a512a/controllers/vrg_volrep.go#L615-L622), we do not call  `skipPVC`. So the deletion will increment the completion count and ignore the PVC. The final result would be that all PVCs are processed (because they would in the end be not protected by the VRG finalizer) even if they are not deleted (yet).

Is there something I am missing?

NOTE: Still to run e2e, so will report back on what actually happens in case something was missed.

Reply:
> In what case, we move the VRG to secondary but the workload is not deleted? IOW, what are we solving?

For StatefulSets, when the STS is deleted the default is that PVCs are not deleted. In this case, for failover action we do not wait for the PVC to be deleted before proceeding to process it as Secondary.

The above would hence result in bound PVCs remaining on the cluster post cleanup. If a future failover were to be attempted to this cluster, it would fail as the PV exists (so s3 restore of the PV would fail). Which is the latter part of the fix, to ensure the PVC is bound to a PV that we want to restore, and if so proceed without errors.

With the second change to not restore PVs, on a cluster where PVCs were not deleted, VRG can become primary. This can happen only on failover, as for relocate we would still check PVC is deleted before processing VRG as secondary.

So overall, we can support STS failovers multiple times across clusters, or cases where PVCs are not deleted by the workload. For relocate we still need the PVCs deleted, to process them as Secondary and report DataReady etc. to make relocation progress (at least as of now).

Reply:
> NOTE: Still to run e2e, so will report back on what actually happens in case something was missed.

Phew! e2e completed. Behaves as expected, multiple failover/relocate with and without deleting PVCs for stateful sets ([example](https://github.com/ShyamsundarR/ocm-ramen-samples/commit/e9dcb4c0a8946f1d2eaa13c9a3a0c8911a852775))

Reply:
I still wonder how the PVC is not protected by VRG given that we only remove the `pvcVRFinalizerProtected` during deletion of the VRG, which comes later than [this check](https://github.com/RamenDR/ramen/blob/acf2e04a81e80a511d387a09e55a7245134a512a/controllers/vrg_volrep.go#L615-L622). But given that you e2e test passed multiple times, then most likely I mis-traced it.

Reply:
Discussed with Benamar, the check in skip is to skip if the PVC was not yet protected by the finalizer, which is what we check ahead in deletion. So currently we both understand that this is safe to do in delete workflow.